File: src/main/java/org/funivan/lologa/algo/ga/genome/value/PossibleMovesValue.java
package org.funivan.lologa.algo.ga.genome.value;

import org.funivan.lologa.algo.find.multiple.PossibleMoves;
import org.funivan.lologa.algo.gameplay.GameplayInterface;
import org.funivan.lologa.tiles.TilesInterface;

public class PossibleMovesValue implements ValueInterface {

    private final PossibleMoves possibleMoves;

    public PossibleMovesValue(GameplayInterface gameplay) {
        this.possibleMoves = new PossibleMoves(gameplay.minimum());
    }

    @Override
    public String type() {
        return "possible_moves";
    }

    @Override
    public Double value(TilesInterface tiles) {
        return (double) this.possibleMoves.handle(tiles).size();
    }
}

File: src/main/java/org/funivan/lologa/algo/ga/genome/value/PossibleMovesValue.kt
package org.funivan.lologa.algo.ga.genome.value

import org.funivan.lologa.algo.find.multiple.PossibleMoves
import org.funivan.lologa.algo.gameplay.GameplayInterface
import org.funivan.lologa.tiles.TilesInterface

class PossibleMovesValue(gameplay: GameplayInterface) : ScoreInterface {

    private val possibleMoves: PossibleMoves

    init {
        this.possibleMoves = PossibleMoves(gameplay.minimum())
    }

    override fun type(): String {
        return "possible_moves"
    }

    override fun value(original: TilesInterface, tiles: TilesInterface): Double {
        return this.possibleMoves.perform(tiles).size().toDouble()
    }
}
